### PR TITLE
feat(consensus): add block_sync reactor

### PR DESCRIPTION
**Pull request by nonvis**

### CODE REVIEW

File: src/noir/consensus/consensus_reactor.cpp [hunk 1]
```diff
@@ -169,74 +169,70 @@ void consensus_reactor::process_peer_msg(p2p::envelope_ptr info) {
...
+      return;
 
-      auto rs = cs_state->get_round_state();
-      auto prs = ps->get_round_state();
+    auto rs = cs_state->get_round_state();
+    auto prs = ps->get_round_state();
 
+    int index{};
+    bool ok{false};
+    if (rs->proposal_block_parts->has_header(prs->proposal_block_part_set_header))
```
Comment (lazka33) [inline src/noir/consensus/consensus_reactor.cpp]: please add a curly brace at here

File: src/noir/consensus/consensus_reactor.h [hunk 2]
```diff
@@ -58,7 +62,15 @@ struct consensus_reactor {
     cs_state->on_start();
   }
 
-  void on_stop() {}
+  void on_stop() {
+    ilog("stopping cs_reactor...");
+    for (auto peer : peers) {
+      ilog(fmt::format("closing peer={}", peer.first));
```
Comment (xedonman) [inline src/noir/consensus/consensus_reactor.h]: I think we can check (is_running == true)

Comment (nonvis) [reply src/noir/consensus/consensus_reactor.h]: got it

File: src/noir/consensus/consensus_reactor.cpp (hunk 1 above)
Comment (nonvis) [reply src/noir/consensus/consensus_reactor.cpp]: got it

**Review by xedonman — COMMENTED** [review]
LGTM

**Review by lazka33 — APPROVED** [review]

**Pull request merged**